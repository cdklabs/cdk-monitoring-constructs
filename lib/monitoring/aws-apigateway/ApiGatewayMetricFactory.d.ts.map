{"version":3,"file":"ApiGatewayMetricFactory.d.ts","sourceRoot":"","sources":["ApiGatewayMetricFactory.ts"],"names":[],"mappings":"AAAA,OAAO,EAAE,WAAW,EAAE,MAAM,4BAA4B,CAAC;AACzD,OAAO,EAAE,aAAa,EAAE,MAAM,4BAA4B,CAAC;AAE3D,OAAO,EAGL,WAAW,EACX,aAAa,EAEb,qBAAqB,EACtB,MAAM,cAAc,CAAC;AAItB,MAAM,WAAW,4BAA4B;IAC3C;;OAEG;IACH,QAAQ,CAAC,GAAG,EAAE,WAAW,CAAC;IAC1B;;OAEG;IACH,QAAQ,CAAC,QAAQ,CAAC,EAAE,MAAM,CAAC;IAC3B;;OAEG;IACH,QAAQ,CAAC,SAAS,CAAC,EAAE,MAAM,CAAC;IAC5B;;OAEG;IACH,QAAQ,CAAC,WAAW,CAAC,EAAE,MAAM,CAAC;IAC9B;;OAEG;IACH,QAAQ,CAAC,iBAAiB,CAAC,EAAE,OAAO,CAAC;IACrC;;OAEG;IACH,QAAQ,CAAC,qBAAqB,CAAC,EAAE,qBAAqB,CAAC;CACxD;AAED,qBAAa,uBAAuB;IAClC,SAAS,CAAC,QAAQ,CAAC,aAAa,EAAE,aAAa,CAAC;IAChD,SAAS,CAAC,QAAQ,CAAC,iBAAiB,EAAE,OAAO,CAAC;IAC9C,SAAS,CAAC,QAAQ,CAAC,qBAAqB,EAAE,qBAAqB,CAAC;IAChE,SAAS,CAAC,QAAQ,CAAC,aAAa,EAAE,aAAa,CAAC;gBAG9C,aAAa,EAAE,aAAa,EAC5B,KAAK,EAAE,4BAA4B;IAcrC;;OAEG;IACH,SAAS;IAUT,oBAAoB;IAUpB,qBAAqB;IAWrB,mBAAmB;IAWnB,kBAAkB;IAUlB,mBAAmB;IAWnB,kBAAkB;IAUlB;;OAEG;IACH,wBAAwB;IAIxB;;OAEG;IACH,wBAAwB;IAIxB;;OAEG;IACH,wBAAwB;IAIxB,qBAAqB,CAAC,WAAW,EAAE,WAAW;CAW/C"}