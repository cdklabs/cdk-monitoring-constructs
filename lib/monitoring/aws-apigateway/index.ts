export * from "./ApiGatewayMetricFactory";
export * from "./ApiGatewayMonitoring";
