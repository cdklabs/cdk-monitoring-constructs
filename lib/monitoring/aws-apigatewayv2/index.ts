export * from "./ApiGatewayV2HttpApiMetricFactory";
export * from "./ApiGatewayV2HttpApiMonitoring";
