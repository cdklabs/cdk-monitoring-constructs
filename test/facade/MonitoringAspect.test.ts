import { SynthUtils } from "@monocdk-experiment/assert";
import { Stack } from "monocdk";
import * as apigw from "monocdk/aws-apigateway";
import * as apigwv2 from "monocdk/aws-apigatewayv2";
import * as acm from "monocdk/aws-certificatemanager";

import {
  DefaultDashboardFactory,
  DashboardRenderingPreference,
  MonitoringFacade,
} from "../../lib";

function createDummyMonitoringFacade(stack: Stack): MonitoringFacade {
  return new MonitoringFacade(stack, "MonitoringFacade", {
    alarmFactoryDefaults: {
      alarmNamePrefix: "DummyMonitoring",
      actionsEnabled: true,
    },
    metricFactoryDefaults: {
      namespace: "DummyNamespace",
    },
    dashboardFactory: new DefaultDashboardFactory(stack, "DashboardFactory", {
      dashboardNamePrefix: "DummyDashboard",
      createDashboard: true,
      createSummaryDashboard: true,
      createAlarmDashboard: true,
      renderingPreference: DashboardRenderingPreference.INTERACTIVE_ONLY,
    }),
  });
}

describe("MonitoringAspect", () => {
  test("ACM", () => {
    // GIVEN
    const stack = new Stack();
    const facade = createDummyMonitoringFacade(stack);
    new acm.Certificate(stack, "DummyCertificate", {
      domainName: "www.monitoring.cdk",
    });

    // WHEN
    facade.monitorScope(stack);

    // THEN
    expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
  });

  test("API Gateway", () => {
    // GIVEN
    const stack = new Stack();
    const facade = createDummyMonitoringFacade(stack);
    const api = new apigw.RestApi(stack, "DummyRestApi");
    api.root.addMethod("ANY");

    // WHEN
    facade.monitorScope(stack);

    // THEN
    expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
  });

  test("API Gateway V2", () => {
    // GIVEN
    const stack = new Stack();
    const facade = createDummyMonitoringFacade(stack);
    new apigwv2.HttpApi(stack, "DummyHttpApi", {
      apiName: "DummyHttpApi",
    });

    // WHEN
    facade.monitorScope(stack);

    // THEN
    expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
  });
});
