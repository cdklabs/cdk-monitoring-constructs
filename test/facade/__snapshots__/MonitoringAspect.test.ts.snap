// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonitoringAspect ACM 1`] = `
Object {
  "Resources": Object {
    "DashboardFactoryAlarmDashboard6286FAD3": Object {
      "Properties": Object {
        "DashboardBody": "{\\"start\\":\\"-PT8H\\",\\"widgets\\":[]}",
        "DashboardName": "DummyDashboard-Alarms",
      },
      "Type": "AWS::CloudWatch::Dashboard",
    },
    "DashboardFactoryDashboard3E20AD6E": Object {
      "Properties": Object {
        "DashboardBody": Object {
          "Fn::Join": Array [
            "",
            Array [
              "{\\"start\\":\\"-PT8H\\",\\"widgets\\":[{\\"type\\":\\"text\\",\\"width\\":24,\\"height\\":1,\\"x\\":0,\\"y\\":0,\\"properties\\":{\\"markdown\\":\\"### Certificate **Default/DummyCertificate**\\"}},{\\"type\\":\\"metric\\",\\"width\\":6,\\"height\\":5,\\"x\\":0,\\"y\\":1,\\"properties\\":{\\"view\\":\\"timeSeries\\",\\"title\\":\\"Expiration\\",\\"region\\":\\"",
              Object {
                "Ref": "AWS::Region",
              },
              "\\",\\"metrics\\":[[\\"AWS/CertificateManager\\",\\"DaysToExpiry\\",\\"CertificateArn\\",\\"",
              Object {
                "Ref": "DummyCertificate611A1363",
              },
              "\\",{\\"label\\":\\"Days to expiry\\",\\"stat\\":\\"Minimum\\"}]],\\"yAxis\\":{\\"left\\":{\\"min\\":0,\\"label\\":\\"Count\\",\\"showUnits\\":false}}}}]}",
            ],
          ],
        },
        "DashboardName": "DummyDashboard",
      },
      "Type": "AWS::CloudWatch::Dashboard",
    },
    "DashboardFactorySummaryDashboard5F4BC8C5": Object {
      "Properties": Object {
        "DashboardBody": "{\\"start\\":\\"-P14D\\",\\"widgets\\":[]}",
        "DashboardName": "DummyDashboard-Summary",
      },
      "Type": "AWS::CloudWatch::Dashboard",
    },
    "DummyCertificate611A1363": Object {
      "Properties": Object {
        "DomainName": "www.monitoring.cdk",
        "DomainValidationOptions": Array [
          Object {
            "DomainName": "www.monitoring.cdk",
            "ValidationDomain": "cdk",
          },
        ],
        "ValidationMethod": "EMAIL",
      },
      "Type": "AWS::CertificateManager::Certificate",
    },
  },
}
`;

exports[`MonitoringAspect API Gateway 1`] = `
Object {
  "Outputs": Object {
    "DummyRestApiEndpoint3DE3EBDB": Object {
      "Value": Object {
        "Fn::Join": Array [
          "",
          Array [
            "https://",
            Object {
              "Ref": "DummyRestApi328A173A",
            },
            ".execute-api.",
            Object {
              "Ref": "AWS::Region",
            },
            ".",
            Object {
              "Ref": "AWS::URLSuffix",
            },
            "/",
            Object {
              "Ref": "DummyRestApiDeploymentStageprodB7A8F869",
            },
            "/",
          ],
        ],
      },
    },
  },
  "Resources": Object {
    "DashboardFactoryAlarmDashboard6286FAD3": Object {
      "Properties": Object {
        "DashboardBody": "{\\"start\\":\\"-PT8H\\",\\"widgets\\":[]}",
        "DashboardName": "DummyDashboard-Alarms",
      },
      "Type": "AWS::CloudWatch::Dashboard",
    },
    "DashboardFactoryDashboard3E20AD6E": Object {
      "Properties": Object {
        "DashboardBody": Object {
          "Fn::Join": Array [
            "",
            Array [
              "{\\"start\\":\\"-PT8H\\",\\"widgets\\":[{\\"type\\":\\"text\\",\\"width\\":24,\\"height\\":1,\\"x\\":0,\\"y\\":0,\\"properties\\":{\\"markdown\\":\\"### API Gateway Endpoint **DummyRestApi prod**\\"}},{\\"type\\":\\"metric\\",\\"width\\":6,\\"height\\":5,\\"x\\":0,\\"y\\":1,\\"properties\\":{\\"view\\":\\"timeSeries\\",\\"title\\":\\"TPS\\",\\"region\\":\\"",
              Object {
                "Ref": "AWS::Region",
              },
              "\\",\\"metrics\\":[[{\\"label\\":\\"Count/s\\",\\"expression\\":\\"FILL(requests,0) / PERIOD(requests)\\"}],[\\"AWS/ApiGateway\\",\\"Count\\",\\"ApiName\\",\\"DummyRestApi\\",\\"Stage\\",\\"prod\\",{\\"label\\":\\"Count\\",\\"stat\\":\\"Sum\\",\\"visible\\":false,\\"id\\":\\"requests\\"}]],\\"yAxis\\":{\\"left\\":{\\"min\\":0,\\"label\\":\\"Count\\",\\"showUnits\\":false}}}},{\\"type\\":\\"metric\\",\\"width\\":6,\\"height\\":5,\\"x\\":6,\\"y\\":1,\\"properties\\":{\\"view\\":\\"timeSeries\\",\\"title\\":\\"Latency\\",\\"region\\":\\"",
              Object {
                "Ref": "AWS::Region",
              },
              "\\",\\"metrics\\":[[\\"AWS/ApiGateway\\",\\"Latency\\",\\"ApiName\\",\\"DummyRestApi\\",\\"Stage\\",\\"prod\\",{\\"label\\":\\"P50\\",\\"stat\\":\\"p50\\"}],[\\"AWS/ApiGateway\\",\\"Latency\\",\\"ApiName\\",\\"DummyRestApi\\",\\"Stage\\",\\"prod\\",{\\"label\\":\\"P90\\",\\"stat\\":\\"p90\\"}],[\\"AWS/ApiGateway\\",\\"Latency\\",\\"ApiName\\",\\"DummyRestApi\\",\\"Stage\\",\\"prod\\",{\\"label\\":\\"P99\\",\\"stat\\":\\"p99\\"}]],\\"yAxis\\":{\\"left\\":{\\"min\\":0,\\"label\\":\\"ms\\",\\"showUnits\\":false}}}},{\\"type\\":\\"metric\\",\\"width\\":6,\\"height\\":5,\\"x\\":12,\\"y\\":1,\\"properties\\":{\\"view\\":\\"timeSeries\\",\\"title\\":\\"Errors\\",\\"region\\":\\"",
              Object {
                "Ref": "AWS::Region",
              },
              "\\",\\"metrics\\":[[\\"AWS/ApiGateway\\",\\"4XXError\\",\\"ApiName\\",\\"DummyRestApi\\",\\"Stage\\",\\"prod\\",{\\"label\\":\\"4XX Error\\",\\"stat\\":\\"Sum\\"}],[\\"AWS/ApiGateway\\",\\"5XXError\\",\\"ApiName\\",\\"DummyRestApi\\",\\"Stage\\",\\"prod\\",{\\"label\\":\\"5XX Fault\\",\\"stat\\":\\"Sum\\"}]],\\"yAxis\\":{\\"left\\":{\\"min\\":0,\\"label\\":\\"Count\\",\\"showUnits\\":false}}}},{\\"type\\":\\"metric\\",\\"width\\":6,\\"height\\":5,\\"x\\":18,\\"y\\":1,\\"properties\\":{\\"view\\":\\"timeSeries\\",\\"title\\":\\"Errors (rate)\\",\\"region\\":\\"",
              Object {
                "Ref": "AWS::Region",
              },
              "\\",\\"metrics\\":[[\\"AWS/ApiGateway\\",\\"4XXError\\",\\"ApiName\\",\\"DummyRestApi\\",\\"Stage\\",\\"prod\\",{\\"label\\":\\"4XX Error (avg)\\"}],[\\"AWS/ApiGateway\\",\\"5XXError\\",\\"ApiName\\",\\"DummyRestApi\\",\\"Stage\\",\\"prod\\",{\\"label\\":\\"5XX Fault (avg)\\"}]],\\"yAxis\\":{\\"left\\":{\\"min\\":0,\\"label\\":\\"Rate\\",\\"showUnits\\":false}}}}]}",
            ],
          ],
        },
        "DashboardName": "DummyDashboard",
      },
      "Type": "AWS::CloudWatch::Dashboard",
    },
    "DashboardFactorySummaryDashboard5F4BC8C5": Object {
      "Properties": Object {
        "DashboardBody": Object {
          "Fn::Join": Array [
            "",
            Array [
              "{\\"start\\":\\"-P14D\\",\\"widgets\\":[{\\"type\\":\\"text\\",\\"width\\":24,\\"height\\":1,\\"x\\":0,\\"y\\":0,\\"properties\\":{\\"markdown\\":\\"### API Gateway Endpoint **DummyRestApi prod**\\"}},{\\"type\\":\\"metric\\",\\"width\\":8,\\"height\\":6,\\"x\\":0,\\"y\\":1,\\"properties\\":{\\"view\\":\\"timeSeries\\",\\"title\\":\\"TPS\\",\\"region\\":\\"",
              Object {
                "Ref": "AWS::Region",
              },
              "\\",\\"metrics\\":[[{\\"label\\":\\"Count/s\\",\\"expression\\":\\"FILL(requests,0) / PERIOD(requests)\\"}],[\\"AWS/ApiGateway\\",\\"Count\\",\\"ApiName\\",\\"DummyRestApi\\",\\"Stage\\",\\"prod\\",{\\"label\\":\\"Count\\",\\"stat\\":\\"Sum\\",\\"visible\\":false,\\"id\\":\\"requests\\"}]],\\"yAxis\\":{\\"left\\":{\\"min\\":0,\\"label\\":\\"Count\\",\\"showUnits\\":false}}}},{\\"type\\":\\"metric\\",\\"width\\":8,\\"height\\":6,\\"x\\":8,\\"y\\":1,\\"properties\\":{\\"view\\":\\"timeSeries\\",\\"title\\":\\"Latency\\",\\"region\\":\\"",
              Object {
                "Ref": "AWS::Region",
              },
              "\\",\\"metrics\\":[[\\"AWS/ApiGateway\\",\\"Latency\\",\\"ApiName\\",\\"DummyRestApi\\",\\"Stage\\",\\"prod\\",{\\"label\\":\\"P50\\",\\"stat\\":\\"p50\\"}],[\\"AWS/ApiGateway\\",\\"Latency\\",\\"ApiName\\",\\"DummyRestApi\\",\\"Stage\\",\\"prod\\",{\\"label\\":\\"P90\\",\\"stat\\":\\"p90\\"}],[\\"AWS/ApiGateway\\",\\"Latency\\",\\"ApiName\\",\\"DummyRestApi\\",\\"Stage\\",\\"prod\\",{\\"label\\":\\"P99\\",\\"stat\\":\\"p99\\"}]],\\"yAxis\\":{\\"left\\":{\\"min\\":0,\\"label\\":\\"ms\\",\\"showUnits\\":false}}}},{\\"type\\":\\"metric\\",\\"width\\":8,\\"height\\":6,\\"x\\":16,\\"y\\":1,\\"properties\\":{\\"view\\":\\"timeSeries\\",\\"title\\":\\"Errors (rate)\\",\\"region\\":\\"",
              Object {
                "Ref": "AWS::Region",
              },
              "\\",\\"metrics\\":[[\\"AWS/ApiGateway\\",\\"4XXError\\",\\"ApiName\\",\\"DummyRestApi\\",\\"Stage\\",\\"prod\\",{\\"label\\":\\"4XX Error (avg)\\"}],[\\"AWS/ApiGateway\\",\\"5XXError\\",\\"ApiName\\",\\"DummyRestApi\\",\\"Stage\\",\\"prod\\",{\\"label\\":\\"5XX Fault (avg)\\"}]],\\"yAxis\\":{\\"left\\":{\\"min\\":0,\\"label\\":\\"Rate\\",\\"showUnits\\":false}}}}]}",
            ],
          ],
        },
        "DashboardName": "DummyDashboard-Summary",
      },
      "Type": "AWS::CloudWatch::Dashboard",
    },
    "DummyRestApi328A173A": Object {
      "Properties": Object {
        "Name": "DummyRestApi",
      },
      "Type": "AWS::ApiGateway::RestApi",
    },
    "DummyRestApiANYCDDCFD86": Object {
      "Properties": Object {
        "AuthorizationType": "NONE",
        "HttpMethod": "ANY",
        "Integration": Object {
          "Type": "MOCK",
        },
        "ResourceId": Object {
          "Fn::GetAtt": Array [
            "DummyRestApi328A173A",
            "RootResourceId",
          ],
        },
        "RestApiId": Object {
          "Ref": "DummyRestApi328A173A",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "DummyRestApiAccount722BC8C0": Object {
      "DependsOn": Array [
        "DummyRestApi328A173A",
      ],
      "Properties": Object {
        "CloudWatchRoleArn": Object {
          "Fn::GetAtt": Array [
            "DummyRestApiCloudWatchRoleA16B4AC6",
            "Arn",
          ],
        },
      },
      "Type": "AWS::ApiGateway::Account",
    },
    "DummyRestApiCloudWatchRoleA16B4AC6": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "apigateway.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AmazonAPIGatewayPushToCloudWatchLogs",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "DummyRestApiDeploymentE536A000827e68cce8aade0ba98340f42cbf5e9e": Object {
      "DependsOn": Array [
        "DummyRestApiANYCDDCFD86",
      ],
      "Properties": Object {
        "Description": "Automatically created by the RestApi construct",
        "RestApiId": Object {
          "Ref": "DummyRestApi328A173A",
        },
      },
      "Type": "AWS::ApiGateway::Deployment",
    },
    "DummyRestApiDeploymentStageprodB7A8F869": Object {
      "Properties": Object {
        "DeploymentId": Object {
          "Ref": "DummyRestApiDeploymentE536A000827e68cce8aade0ba98340f42cbf5e9e",
        },
        "RestApiId": Object {
          "Ref": "DummyRestApi328A173A",
        },
        "StageName": "prod",
      },
      "Type": "AWS::ApiGateway::Stage",
    },
  },
}
`;

exports[`MonitoringAspect API Gateway V2 1`] = `
Object {
  "Resources": Object {
    "DashboardFactoryAlarmDashboard6286FAD3": Object {
      "Properties": Object {
        "DashboardBody": "{\\"start\\":\\"-PT8H\\",\\"widgets\\":[]}",
        "DashboardName": "DummyDashboard-Alarms",
      },
      "Type": "AWS::CloudWatch::Dashboard",
    },
    "DashboardFactoryDashboard3E20AD6E": Object {
      "Properties": Object {
        "DashboardBody": Object {
          "Fn::Join": Array [
            "",
            Array [
              "{\\"start\\":\\"-PT8H\\",\\"widgets\\":[{\\"type\\":\\"text\\",\\"width\\":24,\\"height\\":1,\\"x\\":0,\\"y\\":0,\\"properties\\":{\\"markdown\\":\\"### API Gateway V2 HTTP Endpoint **",
              Object {
                "Ref": "DummyHttpApi8EEBCC85",
              },
              " $default  **\\"}},{\\"type\\":\\"metric\\",\\"width\\":6,\\"height\\":5,\\"x\\":0,\\"y\\":1,\\"properties\\":{\\"view\\":\\"timeSeries\\",\\"title\\":\\"TPS\\",\\"region\\":\\"",
              Object {
                "Ref": "AWS::Region",
              },
              "\\",\\"metrics\\":[[{\\"label\\":\\"TPS\\",\\"expression\\":\\"requests / PERIOD(requests)\\"}],[\\"AWS/ApiGateway\\",\\"Count\\",\\"ApiId\\",\\"",
              Object {
                "Ref": "DummyHttpApi8EEBCC85",
              },
              "\\",\\"Stage\\",\\"$default\\",{\\"label\\":\\"Invocations\\",\\"stat\\":\\"Sum\\",\\"visible\\":false,\\"id\\":\\"requests\\"}]],\\"yAxis\\":{\\"left\\":{\\"min\\":0,\\"label\\":\\"Count\\",\\"showUnits\\":false}}}},{\\"type\\":\\"metric\\",\\"width\\":6,\\"height\\":5,\\"x\\":6,\\"y\\":1,\\"properties\\":{\\"view\\":\\"timeSeries\\",\\"title\\":\\"Latency\\",\\"region\\":\\"",
              Object {
                "Ref": "AWS::Region",
              },
              "\\",\\"metrics\\":[[\\"AWS/ApiGateway\\",\\"Latency\\",\\"ApiId\\",\\"",
              Object {
                "Ref": "DummyHttpApi8EEBCC85",
              },
              "\\",\\"Stage\\",\\"$default\\",{\\"label\\":\\"P50 Latency\\",\\"stat\\":\\"p50\\"}],[\\"AWS/ApiGateway\\",\\"IntegrationLatency\\",\\"ApiId\\",\\"",
              Object {
                "Ref": "DummyHttpApi8EEBCC85",
              },
              "\\",\\"Stage\\",\\"$default\\",{\\"label\\":\\"P50 Integration Latency\\",\\"stat\\":\\"p50\\"}],[\\"AWS/ApiGateway\\",\\"Latency\\",\\"ApiId\\",\\"",
              Object {
                "Ref": "DummyHttpApi8EEBCC85",
              },
              "\\",\\"Stage\\",\\"$default\\",{\\"label\\":\\"P90 Latency\\",\\"stat\\":\\"p90\\"}],[\\"AWS/ApiGateway\\",\\"IntegrationLatency\\",\\"ApiId\\",\\"",
              Object {
                "Ref": "DummyHttpApi8EEBCC85",
              },
              "\\",\\"Stage\\",\\"$default\\",{\\"label\\":\\"P90 Integration Latency\\",\\"stat\\":\\"p90\\"}],[\\"AWS/ApiGateway\\",\\"Latency\\",\\"ApiId\\",\\"",
              Object {
                "Ref": "DummyHttpApi8EEBCC85",
              },
              "\\",\\"Stage\\",\\"$default\\",{\\"label\\":\\"P99 Latency\\",\\"stat\\":\\"p99\\"}],[\\"AWS/ApiGateway\\",\\"IntegrationLatency\\",\\"ApiId\\",\\"",
              Object {
                "Ref": "DummyHttpApi8EEBCC85",
              },
              "\\",\\"Stage\\",\\"$default\\",{\\"label\\":\\"P99 Integration Latency\\",\\"stat\\":\\"p99\\"}]],\\"yAxis\\":{\\"left\\":{\\"min\\":0,\\"label\\":\\"ms\\",\\"showUnits\\":false}}}},{\\"type\\":\\"metric\\",\\"width\\":6,\\"height\\":5,\\"x\\":12,\\"y\\":1,\\"properties\\":{\\"view\\":\\"timeSeries\\",\\"title\\":\\"Errors\\",\\"region\\":\\"",
              Object {
                "Ref": "AWS::Region",
              },
              "\\",\\"metrics\\":[[\\"AWS/ApiGateway\\",\\"4xx\\",\\"ApiId\\",\\"",
              Object {
                "Ref": "DummyHttpApi8EEBCC85",
              },
              "\\",\\"Stage\\",\\"$default\\",{\\"label\\":\\"4xx\\",\\"stat\\":\\"Sum\\"}],[\\"AWS/ApiGateway\\",\\"5xx\\",\\"ApiId\\",\\"",
              Object {
                "Ref": "DummyHttpApi8EEBCC85",
              },
              "\\",\\"Stage\\",\\"$default\\",{\\"label\\":\\"5xx\\",\\"stat\\":\\"Sum\\"}]],\\"yAxis\\":{\\"left\\":{\\"min\\":0,\\"label\\":\\"Count\\",\\"showUnits\\":false}}}},{\\"type\\":\\"metric\\",\\"width\\":6,\\"height\\":5,\\"x\\":18,\\"y\\":1,\\"properties\\":{\\"view\\":\\"timeSeries\\",\\"title\\":\\"Errors (rate)\\",\\"region\\":\\"",
              Object {
                "Ref": "AWS::Region",
              },
              "\\",\\"metrics\\":[[\\"AWS/ApiGateway\\",\\"4xx\\",\\"ApiId\\",\\"",
              Object {
                "Ref": "DummyHttpApi8EEBCC85",
              },
              "\\",\\"Stage\\",\\"$default\\",{\\"label\\":\\"4xx (avg)\\"}],[\\"AWS/ApiGateway\\",\\"5xx\\",\\"ApiId\\",\\"",
              Object {
                "Ref": "DummyHttpApi8EEBCC85",
              },
              "\\",\\"Stage\\",\\"$default\\",{\\"label\\":\\"5xx (avg)\\"}]],\\"yAxis\\":{\\"left\\":{\\"min\\":0,\\"label\\":\\"Rate\\",\\"showUnits\\":false}}}}]}",
            ],
          ],
        },
        "DashboardName": "DummyDashboard",
      },
      "Type": "AWS::CloudWatch::Dashboard",
    },
    "DashboardFactorySummaryDashboard5F4BC8C5": Object {
      "Properties": Object {
        "DashboardBody": Object {
          "Fn::Join": Array [
            "",
            Array [
              "{\\"start\\":\\"-P14D\\",\\"widgets\\":[{\\"type\\":\\"text\\",\\"width\\":24,\\"height\\":1,\\"x\\":0,\\"y\\":0,\\"properties\\":{\\"markdown\\":\\"### API Gateway V2 HTTP Endpoint **",
              Object {
                "Ref": "DummyHttpApi8EEBCC85",
              },
              " $default  **\\"}},{\\"type\\":\\"metric\\",\\"width\\":8,\\"height\\":6,\\"x\\":0,\\"y\\":1,\\"properties\\":{\\"view\\":\\"timeSeries\\",\\"title\\":\\"TPS\\",\\"region\\":\\"",
              Object {
                "Ref": "AWS::Region",
              },
              "\\",\\"metrics\\":[[{\\"label\\":\\"TPS\\",\\"expression\\":\\"requests / PERIOD(requests)\\"}],[\\"AWS/ApiGateway\\",\\"Count\\",\\"ApiId\\",\\"",
              Object {
                "Ref": "DummyHttpApi8EEBCC85",
              },
              "\\",\\"Stage\\",\\"$default\\",{\\"label\\":\\"Invocations\\",\\"stat\\":\\"Sum\\",\\"visible\\":false,\\"id\\":\\"requests\\"}]],\\"yAxis\\":{\\"left\\":{\\"min\\":0,\\"label\\":\\"Count\\",\\"showUnits\\":false}}}},{\\"type\\":\\"metric\\",\\"width\\":8,\\"height\\":6,\\"x\\":8,\\"y\\":1,\\"properties\\":{\\"view\\":\\"timeSeries\\",\\"title\\":\\"Latency\\",\\"region\\":\\"",
              Object {
                "Ref": "AWS::Region",
              },
              "\\",\\"metrics\\":[[\\"AWS/ApiGateway\\",\\"Latency\\",\\"ApiId\\",\\"",
              Object {
                "Ref": "DummyHttpApi8EEBCC85",
              },
              "\\",\\"Stage\\",\\"$default\\",{\\"label\\":\\"P50 Latency\\",\\"stat\\":\\"p50\\"}],[\\"AWS/ApiGateway\\",\\"IntegrationLatency\\",\\"ApiId\\",\\"",
              Object {
                "Ref": "DummyHttpApi8EEBCC85",
              },
              "\\",\\"Stage\\",\\"$default\\",{\\"label\\":\\"P50 Integration Latency\\",\\"stat\\":\\"p50\\"}],[\\"AWS/ApiGateway\\",\\"Latency\\",\\"ApiId\\",\\"",
              Object {
                "Ref": "DummyHttpApi8EEBCC85",
              },
              "\\",\\"Stage\\",\\"$default\\",{\\"label\\":\\"P90 Latency\\",\\"stat\\":\\"p90\\"}],[\\"AWS/ApiGateway\\",\\"IntegrationLatency\\",\\"ApiId\\",\\"",
              Object {
                "Ref": "DummyHttpApi8EEBCC85",
              },
              "\\",\\"Stage\\",\\"$default\\",{\\"label\\":\\"P90 Integration Latency\\",\\"stat\\":\\"p90\\"}],[\\"AWS/ApiGateway\\",\\"Latency\\",\\"ApiId\\",\\"",
              Object {
                "Ref": "DummyHttpApi8EEBCC85",
              },
              "\\",\\"Stage\\",\\"$default\\",{\\"label\\":\\"P99 Latency\\",\\"stat\\":\\"p99\\"}],[\\"AWS/ApiGateway\\",\\"IntegrationLatency\\",\\"ApiId\\",\\"",
              Object {
                "Ref": "DummyHttpApi8EEBCC85",
              },
              "\\",\\"Stage\\",\\"$default\\",{\\"label\\":\\"P99 Integration Latency\\",\\"stat\\":\\"p99\\"}]],\\"yAxis\\":{\\"left\\":{\\"min\\":0,\\"label\\":\\"ms\\",\\"showUnits\\":false}}}},{\\"type\\":\\"metric\\",\\"width\\":8,\\"height\\":6,\\"x\\":16,\\"y\\":1,\\"properties\\":{\\"view\\":\\"timeSeries\\",\\"title\\":\\"Errors (rate)\\",\\"region\\":\\"",
              Object {
                "Ref": "AWS::Region",
              },
              "\\",\\"metrics\\":[[\\"AWS/ApiGateway\\",\\"4xx\\",\\"ApiId\\",\\"",
              Object {
                "Ref": "DummyHttpApi8EEBCC85",
              },
              "\\",\\"Stage\\",\\"$default\\",{\\"label\\":\\"4xx (avg)\\"}],[\\"AWS/ApiGateway\\",\\"5xx\\",\\"ApiId\\",\\"",
              Object {
                "Ref": "DummyHttpApi8EEBCC85",
              },
              "\\",\\"Stage\\",\\"$default\\",{\\"label\\":\\"5xx (avg)\\"}]],\\"yAxis\\":{\\"left\\":{\\"min\\":0,\\"label\\":\\"Rate\\",\\"showUnits\\":false}}}}]}",
            ],
          ],
        },
        "DashboardName": "DummyDashboard-Summary",
      },
      "Type": "AWS::CloudWatch::Dashboard",
    },
    "DummyHttpApi8EEBCC85": Object {
      "Properties": Object {
        "Name": "DummyHttpApi",
        "ProtocolType": "HTTP",
      },
      "Type": "AWS::ApiGatewayV2::Api",
    },
    "DummyHttpApiDefaultStage62F2901C": Object {
      "Properties": Object {
        "ApiId": Object {
          "Ref": "DummyHttpApi8EEBCC85",
        },
        "AutoDeploy": true,
        "StageName": "$default",
      },
      "Type": "AWS::ApiGatewayV2::Stage",
    },
  },
}
`;
